authenticators/remoteuser/remoteuser.go

```diff
@@ -0,0 +1,152 @@
+// Package remoteuser provides an authentication system with login delegation
+//
+// The user's identity validation is delegated to a front-end. This authenticator
+// simply consumes the value of the X-Remote-User HTTP header and generates a signed
+// authentication token with the same ACLs for all users.
+//
+// In the future, per-user ACLs/roles/etc could even be part of other
+// headers. Alternatively, there could also be a local DB here (as in
+// the userlist authenticator). Or a even a combination of both!
```

Comment:
Do you think maybe we can just add a flag to the userlist one, something like a default username when none is given.

Setting that option would trigger the userlist one to accept a empty login and then look for this header, if the header is set it would use the default username to find a user definition and use that as a template.

That feels like less code and probably a bit easier to handle?